pkg/distributor/distributor.go

```diff
@@ -556,6 +556,16 @@ func (d *Distributor) Push(ctx context.Context, req *logproto.PushRequest) (*log
 				continue
 			}
 
+			if ok, until := d.ShouldBlockPolicy(lbs, tenantID); ok {
```

Comment:
Similarly to how we enforce labels, I think this one should also replace the `d.validator.ShouldBlockIngestion` down below which enforces the per-tenant block.
https://github.com/grafana/loki/blob/9f9622d6cdcad871bf55b6352572fa8e92642a52/pkg/distributor/distributor.go#L652-L653